contracts/account/paymaster/PaymasterSigner.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {PackedUserOperation} from "@openzeppelin/contracts/interfaces/draft-IERC4337.sol";
+import {ERC4337Utils} from "@openzeppelin/contracts/account/utils/draft-ERC4337Utils.sol";
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import {PaymasterCore} from "./PaymasterCore.sol";
+import {AbstractSigner} from "../../utils/cryptography/AbstractSigner.sol";
+
+/**
+ * @dev Extension of {PaymasterCore} that adds signature validation. See {SignerECDSA}, {SignerP256} or {SignerRSA}.
+ *
+ * Example of usage:
+ *
+ * ```solidity
+ * contract MyPaymasterECDSASigner is PaymasterSigner, SignerECDSA {
+ *     constructor() EIP712("MyPaymasterECDSASigner", "1") {
+ *       // Will revert if the signer is already initialized
+ *       _initializeSigner(signerAddr);
+ *     }
+ * }
+ * ```
+ */
+abstract contract PaymasterSigner is PaymasterCore, AbstractSigner, EIP712 {
+    using ERC4337Utils for *;
+
+    bytes32 internal constant _USER_OPERATION_REQUEST =
+        keccak256(
+            "UserOperationRequest(address sender,uint256 nonce,bytes initCode,bytes callData,bytes32 accountGasLimits,uint256 paymasterVerificationGasLimit,uint256 preVerificationGas,bytes32 gasFees,uint48 validAfter,uint48 validUntil)"
```

Comment:
I'd reorder it so that it matches more closely the `PackedUserOperation` in `Account`
```suggestion
            "UserOperationRequest(address sender,uint256 nonce,bytes initCode,bytes callData,bytes32 accountGasLimits,uint256 preVerificationGas,bytes32 gasFees,uint256 paymasterVerificationGasLimit,uint48 validAfter,uint48 validUntil)"
```

Reply:
> the paymasterPostOpGasLimit, that is missing.

Yes, I don't see how a user can leverage this field to do any harm. The gas usage is in control of the paymaster itself.

If we decide to add it, I would include it in the `UserOperationRequest` type definition instead of the second option with `paymasterAndDataWithoutSig`. I find that last one problematic if anyone wants to customize the format of the `paymasterData` further.

Reply:
>Would you think it's fine if we include paymasterPostOpGasLimit in the signature hash?

Its less generic, but yes that would be equivalent to "sign everything"